labman/db/plate.py

```diff
@@ -268,6 +336,28 @@ def quantification_process(self):
                 return process_module.QuantificationProcess(res[0][0])
             return None
 
+    @property
+    def duplicates(self):
+        """Get the wells with duplicated samples
+
+        Returns
+        -------
+        dict of {str: list of wells}
+            The duplicated wells, keyed by the sample id
+        """
+        with sql_connection.TRN as TRN:
+            sql = """SELECT sample_id, array_agg(well_id ORDER BY well_id)
+                     FROM qiita.well
+                        JOIN qiita.composition USING (container_id)
+                        JOIN qiita.sample_composition USING (composition_id)
+                     WHERE sample_id IS NOT NULL AND plate_id = %s
+                     GROUP BY sample_id
+                     HAVING array_length(array_agg(well_id), 1) > 1"""
+            TRN.add(sql, [self.id])
+            res = {sample_id: [container_module.Well(w) for w in wells]
```

Comment:
Why do this versus getting everything in a single command. In other words, I'm not sure in the benefit of doing this vs a larger SQL with array_agg

Reply:
The SQL is already doing the array_agg, this is actually construction the object, rather than just returning the ID.

Reply:
K, thanks.